packages/@cdktf/provider-generator/lib/get/generator/constants/provider-namespaces.ts

```diff
@@ -0,0 +1,614 @@
+type Prefix = string;
+export type ResourceNamespace = {
+  name: string;
+  comment: string;
+  additionalPrefix?: Prefix[];
+};
+
+// We need to order these so that the longest prefixes come first (e.g. acmpa before acm)
```

Comment:
I wouldn't want to only rely on key order here – maybe sorting the keys in reverse alphabetical order before lookup could ensure this behaviour as well?

Reply:
Are you sure this is going to make it clearer? I feel like while it's not great either way, but this way it's at least obvious

Reply:
To me a comment like "the longest matching prefix will determine the namespace" would make that clear while being safer than relying on order (which I'm unsure if it is guaranteed at all across Node.js versions 🤔).